haystack_experimental/components/generators/chat/openai.py

```diff
@@ -447,7 +447,10 @@ def _prepare_api_call(  # noqa: PLR0913
             openai_tools = [
                 {
                     "type": "function",
-                    "function": {**t.tool_spec, "strict": tools_strict},
+                    "function": {
+                        **t.tool_spec,
+                        **({"strict": tools_strict} if tools_strict else {}),
```

Comment:
```suggestion
                        **({"strict": tools_strict} if tools_strict is not None else {}),
```

Reply:
We should allow setting `tools_strict` to `false` and only set it to `{}` if no value was provided at all.